### DISCUSSION TITLE
Change colors in a (monochrome) PNG

### DISCUSSION THREAD

**Discussion by halloleo** [ImageMagick/ImageMagick#5690] [Help] [answered]
Hi there

I have transparent PNGs like this:

![](https://halloleo.blot.im/_files/computer.png)

How can I use ImageMagick to change the colors in it to, say, MidnightBlue (i.e. pixels which are 100% black become 100% MidnightBlue and gray pixels become the appropriate shade of blue)?

Also, can I use that command then to change the colors to white and its shades (so that the PNG is visible on a black background)?

Many thanks for pointers!

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform?

I have named your input file "transcomp.png". The image is entirely black, with varying transparency.

We can make all the pixels any colour we want, such as blue or white or #abc, like this:
```
magick transcomp.png -fill Blue -colorize 100 x.png
```
This does not change transparency.

**Reply by halloleo** [reply]
Thanks, @snibgo. Really appreciate you chiming in here!

`colorize` with `fill` works like a charm. I wasn't aware that it does the shades correctly. That's exactly what I need. Will read up on it in the doco.

ps: Just added for the record version info to the question.
